### PR TITLE
fix(rust): improve error message in date_range with invalid units

### PR DESCRIPTION
**Pull request by MarcoGorelli**
Clarifying the error message of `date_range` with invalid unit

Just for my understanding, why does this one raise panic exception as opposed to return `Err`?

### CODE REVIEW

File: py-polars/tests/unit/test_datelike.py [hunk 1]
```diff
@@ -539,6 +539,14 @@ def test_date_range() -> None:
     assert result.cast(pl.Utf8)[-1] == "2022-01-01 00:00:59.247379260"
 
 
+def test_range_invalid_unit() -> None:
+    msg = "unit: 'D' not supported. Available units are: 'ns', 'us', 'ms', 's', 'm', 'h', 'd', 'w', 'mo', 'y', 'i'"
```
Comment (alexander-beedie) [inline py-polars/tests/unit/test_datelike.py:543]: FYI: I probably wouldn't match quite so exactly on the full error string.
`"'D' not supported"` would be sufficient 👍

Comment (MarcoGorelli) [reply py-polars/tests/unit/test_datelike.py]: 😄 sure thing!

**Comment by ritchie46** [comment]
Thank you can you do a rebase?

**Comment by ritchie46** [comment]
Thank you can you do a rebase?

**Pull request merged**